naga/src/back/mod.rs

```diff
@@ -264,6 +264,7 @@ impl crate::TypeInner {
             crate::TypeInner::Image { .. }
             | crate::TypeInner::Sampler { .. }
             | crate::TypeInner::AccelerationStructure { .. } => true,
+            crate::TypeInner::BindingArray { .. } => true,
```

Comment:
I don't think this is right. The documentation for `TypeInner::BindingArray` says:

> The array's element type `base` may be [`Image`], [`Sampler`], or any type that would be permitted for a global in the [`Uniform`] or [`Storage`] address spaces.

So just because `TypeInner` is a `BindingArray` doesn't mean it's a handle.